### PR TITLE
feat: force user to switch to appropriate network

### PR DESCRIPTION
**Pull request by lauchness**
## Description
This PR auto-switches user to the appropriate network.

It allows developers to control what the desired network is via the `DesiredNetworkContext` which can be switched between l2 and l1 - allowing for bridging.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
Review Error for ximxim @ 2023-08-15 18:12:59 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

File: src/components/MintDialog/pages/Bridge/NotStarted.tsx [hunk 1]
```diff
@@ -92,7 +95,18 @@ export const NotStarted: FC<NotStartedProps> = ({
         </div>
         <div className="flex flex-col gap-4">
           {wrongChain && switchNetwork ? (
-            <Button onClick={() => switchNetwork(l1.id)}>Switch to L1</Button>
+            <Button
+              onClick={() => {
+                const isCoinbaseBrowser = getIsCoinbaseBrowser()
```
Comment (wilsoncusack) [inline src/components/MintDialog/pages/Bridge/NotStarted.tsx:100]: I think we want to auto do this? So this should be like in a useEffect on page load here? Don't want to have to click if coinbase wallet

Comment (lauchness) [reply src/components/MintDialog/pages/Bridge/NotStarted.tsx:100]: Right - I only had this in here because it was weird in metamask

I'll update this

**Comment by wilsoncusack** [comment]
@lauchness could you rebase again sorry we had a bug fix go in

File: src/components/MintDialog/MintDialog.tsx [hunk 2]
```diff
@@ -187,7 +209,7 @@ export const MintDialog: FC<{ size?: ButtonProps['size'] }> = ({ size }) => {
           <Success
             resetModal={resetModal}
             txHash={txDetails?.hash ?? ''}
-            closeModal={() => setOpen(false)}
+            closeModal={closeModal}
```
Comment (wilsoncusack) [inline src/components/MintDialog/MintDialog.tsx:212]: what's this about?

Comment (lauchness) [reply src/components/MintDialog/MintDialog.tsx:212]: I made a closeModal function that ensures the `desiredNetwork` gets set back to l2 when the dialog gets closed

File: src/components/DesiredNetworkContext/DesiredNetworkContext.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  const { switchNetwork } = useSwitchNetwork()
+
+  const [desiredNetwork, setDesiredNetwork] = React.useState<
+    typeof l1 | typeof l2
+  >(l2)
+
+  const isCoinbaseBrowser = getIsCoinbaseBrowser()
+  const desiredNetworkId = desiredNetwork.id
+
+  React.useEffect(() => {
```
Comment (wilsoncusack) [inline src/components/DesiredNetworkContext/DesiredNetworkContext.tsx:32]: @lauchness please remove re https://github.com/base-org/onchainsummer.xyz/pull/312, this is causing issues in the dapp browser in the coinbase app

Comment (lauchness) [reply src/components/DesiredNetworkContext/DesiredNetworkContext.tsx:32]: This is the only place where the actual `switchNetwork` function gets called to auto-switch networks

**Review by wilsoncusack — APPROVED** [review]

**Pull request merged**